pkg/vm/vm.go

```diff
@@ -1061,7 +1061,25 @@ func (v *VM) execute(ctx *Context, op opcode.Opcode, parameter []byte) (err erro
 
 		v.refs.Add(val)
 
-	case opcode.PACK:
+	case opcode.PACKMAP:
+		n := toInt(v.estack.Pop().BigInt())
+		if n < 0 || n*2 > v.estack.Len() {
+			panic("invalid length")
+		}
+
+		items := make([]stackitem.MapElement, n)
+		for i := 0; i < n; i++ {
+			key := v.estack.Pop().value
+			if err := stackitem.IsValidMapKey(key); err != nil {
```

Comment:
We have `validateMapKey` function, it also checks that `key` is not null, so maybe use it?

Reply:
To be fair, I think it's somewhat outdated, but it makes things somewhat easier at the same time and that's what VM code usually does.